### DISCUSSION TITLE
WebRTC support mobile browsers?

### DISCUSSION THREAD

**Discussion by hiren3897** [bluenviron/mediamtx#1380] [Questions]
Hello,

Thanks for the amazing WebRTC feature!! :tada: 

 I am trying to read the WebRTC stream in mobile browsers but I am facing a WebSocket error

Does RTSP SIMPLE SERVER provide support Websocket connections for mobile browsers?

## BROWSERS list
- iOS - Safari / Chrome / Firefox 
- iPadOS - Safari / Chrome / Firefox
- Android = Chrome/firefox 

I tried to use run the webRTC stream directly in the mobile browsers but it gives me `ws error`

I am using directly your `new Receiver` logic to connect and create an offer from RTC.

Can you please help me?

Thanks in advance!!

**Comment by aler9** [comment]
Hello, mobile browsers are fully supported. Some of them may require HTTPS, therefore you have to generate a key and enable encryption by using:

```
openssl genrsa -out server.key 2048
openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650
```

then in the configuration:

```yml
webrtcEncryption: yes
webrtcServerKey: server.key
webrtcServerCert: server.crt
```

If you want to embed a stream into a page, try using an iframe that points to the server, otherwise start from the `webrtc_index.html` file. WebSocket is fully supported by mobile browsers.

[a comment by an automated account is not shown]
